Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -239,43 +216,47 @@ public class MpcCoreKit {
             rss_comm: rss_comm)
 
         let key_details = try await tkey!.initialize(never_initialize_new_key: false, include_local_metadata_transitions: false)
-        appState.metadataPubKey = try key_details.pub_key.getPublicKey(format: .EllipticCompress)
-        
+        metadataPubKey = try key_details.pub_key.getPublicKey(format: .EllipticCompress)
+
         if key_details.required_shares > 0 {
             try await existingUser()
         } else {
             try await newUser()
         }
-
-        // to add tss pub details to corekit details
-        let finalKeyDetails = try tkey!.get_key_details()
-        let tssTags = try TssModule.get_all_tss_tags(threshold_key: tkey!)
-        if tssTags.isEmpty {
-            throw CoreKitError.noTssTags
-        }
-        let tssTag = try TssModule.get_tss_tag(threshold_key: tkey!)
-
-        let tssPubKey = try await TssModule.get_tss_pub_key(threshold_key: tkey!, tss_tag: tssTag)
-
-        return MpcKeyDetails(tssPubKey: tssPubKey, metadataPubKey: try finalKeyDetails.pub_key.getPublicKey(format: .EllipticCompress), requiredFactors: finalKeyDetails.required_shares, threshold: finalKeyDetails.threshold, shareDescriptions: finalKeyDetails.share_descriptions, totalShares: finalKeyDetails.total_shares, totalFactors: 0)
+        return try await getKeyDetails()
     }
 
     private func existingUser() async throws {
-        guard tkey != nil else {
+        guard let thresholdKey = tkey else {
             throw CoreKitError.invalidTKey
         }
 
-        // try check for hash factor
-        if option.disableHashFactor == false {
-            let factor = try Utilities.getHashedPrivateKey(postboxKey: oauthKey!, clientID: option.Web3AuthClientId)
-            // TODO: factors need to be verified before insertion
-            try await inputFactor(factorKey: factor)
-            factorKey = factor
-        } else {
-            let factor = try await getDeviceFactor()
-            // TODO: factors need to be verified before insertion
-            try await inputFactor(factorKey: factor)
-            factorKey = factor
+        do {
+            // try check for hash factor
+
+            let hashFactor = try Utilities.getHashedPrivateKey(postboxKey: oAuthKey!, clientID: option.web3AuthClientId)
+            let hashFactorPub = try SecretKey(hex: hashFactor).toPublic().serialize(compressed: true)
+            let allFactorPub = try await getAllFactorPubs()
+
+            if option.disableHashFactor == false && allFactorPub.contains(hashFactorPub) {
```

Comment:
we should handle the case when hashedFactor is deleted, not sure if getAllFactorPubs is already handling it, below is the code for how web is handling
<img width="758" alt="Screenshot 2024-12-03 at 10 07 23 AM" src="https://github.com/user-attachments/assets/a354d51e-5d87-41b7-8864-e191e2885386">

Reply:
update: getAllFactorPubs already handles if factor is deleted so this works